src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php

```diff
@@ -0,0 +1,510 @@
+        }
+
+        $cmsCategoryGridFactory = $this->get('prestashop.core.grid.factory.cms_page_category');
+        $cmsCategoryGrid = $cmsCategoryGridFactory->getGrid($filters);
+
+        $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
+
+        return $this->render('@PrestaShop/Admin/Improve/Design/Cms/index.html.twig', [
+            'cmsCategoryGrid' => $gridPresenter->present($cmsCategoryGrid),
+            'cmsPageView' => $viewData,
+        ]);
+    }
+
+    /**
+     * Displays cms category page form and handles create new cms page category logic.
+     *
+     * @AdminSecurity(
+     *     "is_granted('create', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_cms_pages_index",
+     *     redirectQueryParamsToKeep={"cmsCategoryParentId"},
+     *     message="You do not have permission to add this."
+     * )
+     *
+     * @return RedirectResponse
+     */
+    public function createCmsCategoryAction()
+    {
+//        todo: remove legacy parts once form is ready and demoRestricted on post action
+        $legacyLink = $this->getAdminLink('AdminCmsContent', [
+            'addcms_category' => 1,
+        ]);
+
+        return $this->redirect($legacyLink);
+    }
+
+    /**
+     * Displays cms category page form and handles update cms page category logic.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_cms_pages_index",
+     *     redirectQueryParamsToKeep={"cmsCategoryParentId"},
+     *     message="You do not have permission to edit this."
+     * )
+     *
+     * @param int $cmsCategoryId
+     *
+     * @return RedirectResponse
+     */
+    public function editCmsCategoryAction($cmsCategoryId)
+    {
+//        todo: remove legacy parts once form is ready and demoRestricted on post action
+        $legacyLink = $this->getAdminLink('AdminCmsContent', [
+            'id_cms_category' => $cmsCategoryId,
+            'updatecms_category' => 1,
+        ]);
+
+        return $this->redirect($legacyLink);
+    }
+
+    /**
+     * Deletes cms page category and all its children categories.
+     *
+     * @AdminSecurity(
+     *     "is_granted('delete', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_cms_pages_index",
+     *     redirectQueryParamsToKeep={"cmsCategoryParentId"},
+     *     message="You do not have permission to delete this."
+     * )
+     * @DemoRestricted(
+     *     redirectRoute="admin_cms_pages_index",
+     *     redirectQueryParamsToKeep={"cmsCategoryParentId"}
+     * )
+     *
+     * @param int $cmsCategoryId
+     *
+     * @return RedirectResponse
+     */
+    public function deleteCmsCategoryAction($cmsCategoryId)
+    {
+        $cmsCategoryParentId = null;
+        try {
+            /** @var CmsPageCategoryId $cmsCategoryParentId */
+            $cmsCategoryParentId = $this->getCommandBus()->handle(
```

Comment:
I can see why you did it but ... no 😅 . A Command that deletes one category is not supposed to return the parent ID. This does not make sense in the domain. The domain should not solve the app layer (controllers, templating, routing ...) issues.

You can:
- retrieve the parent ID before
or
- return user on root listing

Reply:
yeah, I got your point. :laughing: 
I think I should make another Query command which would return parent id by child id, wdyt?